pkgs/firehose/lib/src/changelog.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+class Changelog {
+  final File file;
+
+  Changelog(this.file);
+
+  bool get exists => file.existsSync();
+
+  String? get latestVersion {
+    var sections = _parseSections();
+    return sections.isEmpty ? null : sections.first.title.substring(3).trim();
+  }
+
+  List<String> get latestChangeEntries {
+    var sections = _parseSections();
+    return sections.isEmpty ? [] : sections.first.entries;
+  }
+
+  List<_Section> _parseSections() {
```

Comment:
[optional] Consider using a `sync*` method here.

Both callers to `_parseSections` read only the first value, so I don't think we need to find the entire list eagerly.

Could also consider changing it to `_Section? _parseFirstSection()`